File: java/src/test/java/com/dddeurope/niuz/fixtures/TestFacade.java
package com.dddeurope.niuz.fixtures;

import com.dddeurope.niuz.commands.CommandDispatcher;
import com.dddeurope.niuz.finance.ContractRepository;
import com.dddeurope.niuz.finance.PaymentDto;
import com.dddeurope.niuz.finance.PaymentService;
import com.dddeurope.niuz.hr.AuthorService;
import com.dddeurope.niuz.hr.HireAuthor;
import com.dddeurope.niuz.newsroom.NewsroomService;
import com.dddeurope.niuz.website.TeaserDto;
import com.dddeurope.niuz.newsroom.ArticleRepository;
import com.dddeurope.niuz.newsroom.AuthorRepository;
import com.dddeurope.niuz.finance.PaymentRepository;
import com.dddeurope.niuz.website.TeaserRepository;
import com.dddeurope.niuz.website.TeaserService;

import java.util.List;

public class TestFacade {
    private final AuthorService authorService;
    private final NewsroomService newsroomService;
    private final TeaserService teaserService;
    private final PaymentService paymentService;

    public TestFacade() {
        AuthorRepository authors = new InMemoryAuthors();
        ArticleRepository articles = new InMemoryArticles();
        TeaserRepository teasers = new InMemoryTeasers();
        PaymentRepository payments = new InMemoryPayments();
        ContractRepository contracts = new InMemoryContractRepository();
        InMemoryEventBus eventBus = new InMemoryEventBus();

        authorService = new AuthorService(eventBus);
        newsroomService = new NewsroomService(authors, articles, eventBus, eventBus);
        teaserService = new TeaserService(teasers, eventBus);
        paymentService = new PaymentService(contracts, payments, eventBus);
    }

    public void hire(String authorId, String authorName, String bankAccount, String contractType, int rate) {
        new CommandDispatcher(authorService).dispatch(new HireAuthor(authorId, authorName, contractType, rate, bankAccount));
    }

    public void submit(String articleId, String authorId, String headline) {
        newsroomService.submit(articleId, authorId, headline);
    }

    public void publish(String articleId) {
        newsroomService.publish(articleId);
    }

    public List<TeaserDto> get(String page) {
        return teaserService.get(page);
    }

    public List<PaymentDto> getByBankAccount(String bankAccount) {
        return paymentService.getByBankAccount(bankAccount);
    }
}

File: dotnet/niuz.application.test/fixtures/TestFacade.cs
using System.Collections.Generic;
using niuz.application.commands;
using niuz.application.finance;
using niuz.application.hr;
using niuz.application.newsroom;
using niuz.application.website;

namespace niuz.application.fixtures
{
    public class TestFacade
    {
        private readonly AuthorService authorService;
        private readonly NewsRoomService newsRoomService;
        private readonly TeaserService teaserService;
        private readonly PaymentService paymentService;
        private CommandDispatcher commandDispatcher;

        public TestFacade()
        {
            var authors = new InMemoryAuthors();
            var articles = new InMemoryArticles();
            var teasers = new InMemoryTeasers();
            var payments = new InMemoryPayments();
            var eventBus = new InMemoryEventBus();

            authorService = new AuthorService(eventBus);
            newsRoomService = new NewsRoomService(authors, articles, eventBus, eventBus);
            teaserService = new TeaserService(teasers, eventBus);
            paymentService = new PaymentService(new InMemoryContracts(), payments, eventBus);
            commandDispatcher = new CommandDispatcher(authorService);
        }

        public void Hire(string authorId, string authorName, string bankAccount, string contractType, int rate)
        {
            commandDispatcher.Dispatch(new HireAuthor(authorId, authorName, contractType, rate, bankAccount));
        }

        public void Submit(string articleId, string authorId, string headline)
        {
            newsRoomService.Submit(articleId, authorId, headline);
        }

        public void Publish(string articleId)
        {
            newsRoomService.Publish(articleId);
        }

        public IEnumerable<TeaserDto> Get(string page)
        {
            return teaserService.Get(page);
        }

        public IEnumerable<PaymentDto> GetByBankAccount(string bankAccount)
        {
            return paymentService.GetByBankAccount(bankAccount);
        }
    }
}